invenio_search_ui/searchconfig.py

```diff
@@ -0,0 +1,274 @@
+            "filters": self.initial_filters,
+        }
+
+    @property
+    def searchApi(self):
+        """Generate searchAPI configuration."""
+        return {
+            "axios": {
+                "url": self.endpoint,
+                "withCredentials": True,
+                "headers": self.headers,
+            },
+            "invenio": {
+                "requestSerializer":
+                    "InvenioRecordsResourcesRequestSerializer",
+            }
+        }
+
+    @property
+    def layoutOptions(self):
+        """Generate the Layout Options.
+
+        :returns: A dict with the options for React-SearchKit JS.
+        """
+        return {
+            'listView': self.list_view,
+            'gridView': self.grid_view
+        }
+
+    @property
+    def sortOptions(self):
+        """Format sort options to be used in React-SearchKit JS.
+
+        :returns: A list of dicts with sorting options for React-SearchKit JS.
+        """
+        return list(self.sort) if self.sort is not None else []
+
+    @property
+    def aggs(self):
+        """Format the aggs configuration to be used in React-SearchKit JS.
+
+        :returns: A list of dicts for React-SearchKit JS.
+        """
+        return list(self.facets) if self.facets is not None else []
+
+    @property
+    def paginationOptions(self):
+        """Format the pagination options to be used in React-SearchKit JS."""
+        if not getattr(self, 'default_size') or \
+                self.default_size not in self.pagination_options:
+            raise ValueError(
+                'Parameter default_size should be part of pagination_options')
+        return {
+                "resultsPerPage": [
+                    {"text": str(option), "value": option}
+                    for option in self.pagination_options
+                ],
+                "defaultValue": self.default_size,
+            }
+
+    @property
+    def defaultSortingOnEmptyQueryString(self):
+        """Defines the default sorting options when there is no query."""
+        return {
+            "sortBy": self.sort.default_no_query,
+        }
+
+    @classmethod
+    def generate(cls, options, **kwargs):
+        """Create JSON config for React-Searchkit."""
+        generator_object = cls(options)
+        config = {
+            "appId": generator_object.appId,
+            "initialQueryState": generator_object.initialQueryState,
+            "searchApi": generator_object.searchApi,
+            "sortOptions": generator_object.sortOptions,
+            "aggs": generator_object.aggs,
+            "layoutOptions": generator_object.layoutOptions,
+            "sortOrderDisabled": True,
+            "paginationOptions": generator_object.paginationOptions,
+            "defaultSortingOnEmptyQueryString":
+                generator_object.defaultSortingOnEmptyQueryString
+
+        }
+        config.update(kwargs)
+        return config
+
+
+#
+# Application state context processors
```

Comment:
I would maybe change this comment to 
```suggestion
# Application state context generators, to be used in context processors
```

since we don't actually define any context processor here